src/ripple/protocol/impl/STTx.cpp

```diff
@@ -261,7 +266,10 @@ std::pair<bool, std::string> STTx::checkSingleSign () const
     bool validSig = false;
     try
     {
-        bool const fullyCanonical = (getFlags() & tfFullyCanonicalSig);
+        bool const fullyCanonical =
+            (getFlags() & tfFullyCanonicalSig) ||
+            (requireCanonicalSig != RequireFullyCanonicalSig::no);
```

Comment:
I'd prefer `requireCanonicalSig == RequireFullyCanonicalSig::yes` here (and below in **`checkMultiSign`**). It makes it easier to read and inverting the check just makes it harder (at least for me) to reason about this.

Reply:
I hear the argument.  Personally, I picked `!= no` because I know that it's possible for a `bool` to have values other than 0 and 1.  So in case there's something unexpected in the `bool` it will set `fullyCanonical == true`.  Using `== yes` doesn't give that guarantee.

But I don't feel strongly about it.  The case is pretty unlikely.